apis/flowmetrics/v1alpha1/flowmetric_webhook.go

```diff
@@ -0,0 +1,70 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// log is for logging in this package.
+var flowmetriclog = logf.Log.WithName("flowmetric-resource")
+
+func (r *FlowMetric) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// based on the docs : https://docs.openshift.com/container-platform/latest/observability/network_observability/json-flows-format-reference.html.
+var validFieldsNames = []string{
+	"Bytes", "DnsErrno", "DnsFlags", "DnsFlagsResponseCode", "DnsId", "DnsLatencyMs",
+	"Dscp", "DstAddr", "DstK8S_HostIP", "DstK8S_HostName", "DstK8S_Name", "DstK8S_Namespace", "DstK8S_OwnerName",
+	"DstK8S_OwnerType", "DstK8S_Type", "DstK8S_Zone", "DstMac", "DstPort", "Duplicate", "Flags", "FlowDirection",
+	"IcmpCode", "IcmpType", "IfDirection", "Interface", "K8S_ClusterName", "K8S_FlowLayer", "Packets",
+	"PktDropBytes", "PktDropLatestDropCause", "PktDropLatestFlags", "PktDropLatestState", "PktDropPackets",
+	"Proto", "SrcAddr", "SrcK8S_HostIP", "SrcK8S_HostName", "SrcK8S_Name", "SrcK8S_Namespace", "SrcK8S_OwnerName", "SrcK8S_OwnerType",
+	"SrcK8S_Type", "SrcK8S_Zone", "SrcMac", "SrcPort", "TimeFlowEndMs", "TimeFlowStartMs", "TimeReceived", "TimeFlowRttNs",
+}
```

Comment:
Instead of this hard-coded list, we can leverage the embedded frontend config which already contains the fields list: see here https://github.com/netobserv/network-observability-operator/blob/main/controllers/consoleplugin/consoleplugin_objects.go#L414-L415

If we create a `config.GetStaticConfig` func in consoleplugin package, we can get that same fields list, and we keep a single source of truth

Reply:
cc @OlivierCazade I guess the same approach would be valid on your task